WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2056,6 +2055,12 @@ public int getCount() {
         }
     }
 
+    private boolean shouldShowGutenbergEditor() {
+        return AppPrefs.isGutenbergEditorEnabled() &&
+               (mIsNewPost || TextUtils.isEmpty(mPost.getContent()) ||
```

Comment:
> TextUtils.isEmpty(mPost.getContent())

Which user flow does this condition capture?

I wonder, can we just check for `mIsNewPost` and `contentContainsGutenbergBlocks` in `onCreate` and set a `mShowGutenbergEditor` flag to true? I'm under the impression that we don't need the `shouldShowGutenbergEditor` to be dynamic and it's OK to be computed once, in the beginning. Am I missing some case?

Reply:
@hypest if we have drafted post without content, we want to open it from Gutenberg, right? so that would be the case for that.
Regarding computed once, I agree, I will see to change that part of the code.

Reply:
> @hypest if we have drafted post without content, we want to open it from Gutenberg, right? so that would be the case for that.

Hmm, isn't that captured by the `mIsNewPost` condition? I can be wrong but it seems that `mIsNewPost` will be true when starting a new post, no?

Reply:
Let me check :) Thanks for idea :)

Reply:
Which flow are you trying and see that debug state @marecar3 ?

Tried both the main bottombar "New post" button and the posts list's FAB and `mIsNewPost` ends up being true.

Reply:
> if we have drafted post without content

Oh, you mean that if we have started the post with Gutenberg left it empty went back and then re-entered it, it won't detect that it's still "new"?

Reply:
Thinking of it, if the user has Gutenberg enabled and the post is empty then it makes sense to open Gutenberg regardless of which editor the empty draft was originally created with.
If we remove this condition, then the a user that:
- has Gutenberg enabled
- then goes edit an empty post
-> should expect to edit in Gutenberg, not Aztec

So I believe the condition should remain here, tl;dr we shouldn't care about the origin of the post when it's empty, but trust the user if they have enabled Gutenberg

wdyt @hypest @marecar3 ?

Reply:
As discussed over Slack, let's follow what wpios is doing atm to avoid creating a task for the iOS team at this hour. So, let's use whatever editor is the preferred/selected one besides Gutenberg.

Reply:
@mzorz I am sharing your thoughts, but let's follow the iOS to not make them extra work.

Reply:
Cool, let's go ahead

Reply:
@mzorz @hypest @jtreanor all comments are fixed.
let me know if we can merge this one.